lib/active_force/sobject.rb

```diff
@@ -215,7 +215,7 @@ def logger_output action, exception, params = {}
     end
 
     def attributes_for_sfdb
-      attrs = self.class.mapping.translate_to_sf(modified_attributes)
+      attrs = self.class.mapping.translate_to_sf(@attributes.values_for_database.slice(*changed))
```

Comment:
Since `modified_attributes` was a public method, I didn't want to change it's behavior.